### PR TITLE
Gridtools 1.1.2 [WIP]

### PR DESCRIPTION
**Pull request by lukasm91**
Enables compilation with HIP and Clang-CUDA.

See #1361 #1387 #1389

### CODE REVIEW

File: docs_src/manuals/getting_started/code/CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,10 @@ cmake_minimum_required(VERSION 3.14.5)
 
 project(GridTools-laplacian LANGUAGES CXX)
 
-find_package(GridTools 1.1.0 REQUIRED)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+find_package(GridTools 2.0.0 REQUIRED)
```
Comment (fthaler) [inline docs_src/manuals/getting_started/code/CMakeLists.txt]: Should be 1.1.2 or 1.1, I guess

File: include/gridtools/common/timer/timer_cuda.hpp [hunk 2]
```diff
@@ -23,12 +23,16 @@ namespace gridtools {
      */
     class timer_cuda : public timer<timer_cuda> // CRTP
     {
+        struct destroy_event {
+            inline void operator()(cudaEvent_t* ptr) { cudaEventDestroy(*ptr); }
```
Comment (fthaler) [inline include/gridtools/common/timer/timer_cuda.hpp]: `delete ptr` should additionally be there :P

**Comment by havogt** [comment]
Is this more than cherry-picks?

**Comment by havogt** [comment]
Reminder: Don't squash merge cherry-picks!

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch jenkins

**Comment by havogt** [comment]
launch strgrid

**Pull request merged**